src/main/java/mil/dds/anet/database/PositionDao.java

```diff
@@ -176,139 +180,206 @@ public int updateInternal(Position p) {
-                  + "WHERE \"currentPersonUuid\" = :personUuid")
-          .bind("personUuid", personUuid).execute();
+    int numRows = 0;
+    // If the position is already assigned to another person, remove the person from the position
+    numRows += removePersonFromPosition(positionUuid, now);
+    // If the person is already assigned to another position, remove the position from the person
+    numRows += removePositionFromPerson(personUuid, now);
 
-      getDbHandle()
-          .createUpdate("/* positionSetPerson.remove2 */ INSERT INTO \"peoplePositions\" "
-              + "(\"positionUuid\", \"personUuid\", \"createdAt\") "
-              + "VALUES (:positionUuid, NULL, :createdAt)")
-          .bind("positionUuid", currPos.getUuid()).bind("createdAt", DaoUtils.asLocalDateTime(now))
-          .execute();
-    }
+    numRows += updatePersonOfPosition(positionUuid, personUuid, now);
+
+    numRows += getDbHandle()
+        .createUpdate(
+            "/* positionPerson.end */ UPDATE \"peoplePositions\" SET \"endedAt\" = :endedAt "
+                + "WHERE \"positionUuid\" = :positionUuid AND " + " \"endedAt\" IS NULL")
+        .bind("positionUuid", positionUuid)
+        .bind("endedAt", DaoUtils.asLocalDateTime(now.plusMillis(2))).execute();
+
+    numRows += getDbHandle()
+        .createUpdate(
+            "/* personPosition.end */ UPDATE \"peoplePositions\" SET \"endedAt\" = :endedAt "
+                + "WHERE \"personUuid\" = :personUuid AND " + " \"endedAt\" IS NULL")
+        .bind("personUuid", personUuid).bind("endedAt", DaoUtils.asLocalDateTime(now.plusMillis(2)))
+        .execute();
 
-    getDbHandle()
-        .createUpdate("/* positionSetPerson.set1 */ UPDATE positions "
-            + "SET \"currentPersonUuid\" = :personUuid WHERE uuid = :positionUuid")
-        .bind("personUuid", personUuid).bind("positionUuid", positionUuid).execute();
     // GraphQL mutations *have* to return something, so we return the number of inserted rows
-    final int nr = getDbHandle()
-        .createUpdate("/* positionSetPerson.set2 */ INSERT INTO \"peoplePositions\" "
+    numRows += getDbHandle()
+        .createUpdate("/* positionPerson.insert */ INSERT INTO \"peoplePositions\" "
             + "(\"positionUuid\", \"personUuid\", \"createdAt\") "
             + "VALUES (:positionUuid, :personUuid, :createdAt)")
         .bind("positionUuid", positionUuid).bind("personUuid", personUuid)
         // Need to ensure this timestamp is greater than previous INSERT.
-        .bind("createdAt", DaoUtils.asLocalDateTime(now.plusMillis(1))).execute();
+        .bind("createdAt", DaoUtils.asLocalDateTime(now.plusMillis(2))).execute();
+
     // Evict this person from the domain users cache, as their position has changed
     AnetObjectEngine.getInstance().getPersonDao().evictFromCacheByPersonUuid(personUuid);
 
-    return nr;
+    if (numRows > 0) {
+      return 1;
+    }
+    return numRows;
```

Comment:
Otherwise, these statements (here and elsewhere) can be simplified to `return numRows > 0 ? 1 : 0;` .

Reply:
Only where necessary was numRows used and defined as final.